habitat/tasks/rearrange/rearrange_pick_task.py

```diff
@@ -0,0 +1,392 @@
+from habitat.tasks.rearrange.envs.utils import (
+    CacheHelper,
+    get_angle,
+    rearrang_collision,
+    reshape_obs_space,
+)
+from habitat.tasks.rearrange.rearrange_task import RearrangeTask
+
+
+def merge_sim_episode_with_object_config(sim_config, episode):
+    sim_config.defrost()
+    sim_config.ep_info = [episode.__dict__]
+    sim_config.freeze()
+    return sim_config
+
+
+@registry.register_task(name="RearrangePickTask-v0")
+class RearrangePickTask(NavigationTask):
+
+    """
+    Embodied Question Answering Task
+    Usage example:
+        env = habitat.Env(config=eqa_config)
+
+        env.reset()
+
+        for i in range(10):
+            action = sample_non_stop_action(env.action_space)
+            if action["action"] != AnswerAction.name:
+                env.step(action)
+            metrics = env.get_metrics() # to check distance to target
+
+        correct_answer_id = env.current_episode.question.answer_token
+        env.step(
+            {
+                "action": AnswerAction.name,
+                "action_args": {"answer_id": correct_answer_id},
+            }
+        )
+
+        metrics = env.get_metrics()
+    """
+
+    def overwrite_sim_config(self, sim_config, episode):
+        return merge_sim_episode_with_object_config(sim_config, episode)
+
+
+@registry.register_task(name="RearrangePickTask-v1")
+class RearrangePickTaskV1(RearrangeTask):
+
+    """
+    Embodied Question Answering Task
+    Usage example:
+        env = habitat.Env(config=eqa_config)
+
+        env.reset()
+
+        for i in range(10):
+            action = sample_non_stop_action(env.action_space)
+            if action["action"] != AnswerAction.name:
+                env.step(action)
+            metrics = env.get_metrics() # to check distance to target
+
+        correct_answer_id = env.current_episode.question.answer_token
+        env.step(
+            {
+                "action": AnswerAction.name,
+                "action_args": {"answer_id": correct_answer_id},
+            }
+        )
+
+        metrics = env.get_metrics()
+    """
+
+    def overwrite_sim_config(self, sim_config, episode):
+        return merge_sim_episode_with_object_config(sim_config, episode)
+
+    def __init__(self, *args, config, dataset=None, **kwargs):
+        super().__init__(config=config, *args, dataset=dataset, **kwargs)
+        # super().__init__(config, dataset)
+        self.cache = {}
+
+        data_path = dataset.config.DATA_PATH.format(split=dataset.config.SPLIT)
+
+        mtime = osp.getmtime(data_path)
+        cache_name = (
+            str(mtime)
+            + dataset.config.SPLIT
+            + str(self._config.COUNT_OBJ_COLLISIONS)
+        )
+        cache_name += str(self._config.BASE_NOISE)
+        cache_name = cache_name.replace(".", "_")
+
+        fname = data_path.split("/")[-1].split(".")[0]
+
+        self.cache = CacheHelper(
+            "start_pos", cache_name, {}, verbose=True, rel_dir=fname
```

Comment:
```suggestion
            "start_pos", cache_name, {}, verbose=False, rel_dir=fname
```
I'd prefer it to be not verbose unless uses really need to know that something was saved